src/bitops.c

```diff
@@ -813,16 +816,17 @@ void bitcountCommand(client *c) {
                 return;
             }
         }
-        if (isbit) strlen <<= 3;
-        if (start < 0) start = strlen+start;
-        if (end < 0) end = strlen+end;
+        if (isbit) totlen <<= 3;
```

Comment:
I don't understand why we may overflow. Or I miss something?

Reply:
the `totlen <<= 3`.
similar to this assertion we have below:
```c
serverAssert(totlen <= LLONG_MAX >> 3);
```

i.e. we concluded that on 32bit system, the biggest string will never reach these sizes since we use a 64 bit long long.
and on 64 bit in theory we can have an overflow, but in practice it's probably not a real scenario we can be facing (such a huge string), but if there is ever such a case, we rather assert than proceed and possibly corrupt the heap later on.

Reply:
I already assert it at line 801.

Reply:
ohh, missed that.. sorry.